selfhost/typechecker.jakt

```diff
@@ -915,6 +915,7 @@ boxed enum CheckedExpression {
     Boolean(val: bool, span: Span)
     NumericConstant(val: CheckedNumericConstant, span: Span, type_id: TypeId)
     QuotedString(val: String, span: Span)
+    FormatString(val: String, span: Span)
```

Comment:
This is _far_ too "special" to be useful, especially as the current format of these functions is temporary until we get variadic generics
Let's keep the check for literals, and not overfit to a temporary state (split this commit at least)

Reply:
I made it its own commit for now